### ISSUE TITLE
Skip ihooks tests in twisted.test.test_reflect when it isn't available

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5138]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5138|
|Type|task|
|Created|2011-06-06 17:30:13Z|
|Branch|https://github.com/twisted/twisted/tree/ihooks-skip-5138-2|

Twisted emits this Py3k warning:

```
the ihooks module has been removed in Python 3.0
```

for the following test:

```
./twisted/test/test_reflect.py:from ihooks import ModuleImporter
```

<details><summary>Searchable metadata</summary>

```
trac-id__5138 5138
type__task task
reporter__thijs thijs
priority__low low
milestone__Python_3_x Python-3.x
branch__branches_ihooks_skip_5138_2 branches/ihooks-skip-5138-2
branch_author__vperic vperic
status__closed closed
resolution__fixed fixed
component__core core
keywords__py3k py3k
time__1307381413000000 1307381413000000
changetime__1343293933000000 1343293933000000
version__None None
owner__vperic vperic
cc__thijs cc__vperic
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

`test_reflect.py` uses `ModuleImporter` to test that `namedAny` and the associated functions work correctly when `ihooks` is in use.  There's nothing to replace the usage with, because the tests are specifically for `ihooks` support.  The tests could be made conditional and skipped if `ihooks` is not available, but that won't get rid of the Py3k warning.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_3"></a>|@vperic commented|
|-|-|

(In [34583]) Create branch ihooks-skip-5138

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_4"></a>|@vperic commented|
|-|-|

This was simple enough, to quote my commit message:

    The ihooks module has been removed in Python 3, but there is a test to
    make sure it works correctly. Skip this test if ihooks isn't present. This
    does not remove the warning thrown by "-3", but is a viable fix.

[Build results.](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/ihooks-skip-5138)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_5"></a>|@thijstriemstra commented|
|-|-|

Replying to [vperic](#note_4):
> This was simple enough, to quote my commit message:

Wanted to point out that you can add your commit messages to a ticket by including a reference to the ticket nr in your commit message like 'refs [#5138](https://github.com/twisted/twisted/issues/5138)'.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra set owner to @vperic|
|-|-|

Please merge with a .misc news file, thanks.

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_7"></a>|@vperic commented|
|-|-|

(In [34733]) Add topfile for 5138

refs [#5138](https://github.com/twisted/twisted/issues/5138)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra commented|
|-|-|

ping :)

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_9"></a>|@vperic commented|
|-|-|

(In [34964]) Create branch ihooks-skip-5138-2

**Comment by twisted-trac** [comment]
|[<img alt="vperic's avatar" src="https://avatars.githubusercontent.com/u/688281?s=50" width="50" height="50">](https://github.com/vperic)<a name="note_10"></a>|@vperic set status to `closed`|
|-|-|

(In [34966]) Merge ihooks-skip-5138-2: Skip test if ihooks isn't available

Author: vperic
Reviewer: thijs
Fixes: [#5138](https://github.com/twisted/twisted/issues/5138)

The ihooks module has been removed in Python 3, but there is a test to
make sure it works correctly. Skip this test if ihooks isn't present. This
does not remove the warning thrown by "-3", but is a viable fix.
